pkg/deploy/deploy_rp.go

```diff
@@ -186,3 +192,23 @@ func (d *deployer) convertToIPAddressOrRange(ipSlice []string) []mgmtdocumentdb.
 	}
 	return ips
 }
+
+func (d *deployer) rpRemoveFailedScaleset(ctx context.Context) error {
+	d.log.Print("removing failed scaleset")
+
+	// Name is [concat('rp-vmss-', parameters('vmssName'))]; vmssName is d.version
+	vmss, err := d.vmss.Get(ctx, d.config.RPResourceGroupName, "rp-vmss-"+d.version)
```

Comment:
Does this return `nil` for `err` if the VMSS doesn't exist? We need it to not fail if the VMSS does not exist or have this function return the vmss and the error so caller can determine what to do.

Reply:
It would return an error. The error text from the CLI is below:
>(ResourceNotFound) The Resource 'Microsoft.Compute/virtualMachineScaleSets/x' under resource group 'y' was not found. For more details please go to https://aka.ms/ARMResourceNotFoundFix

`rpRemoveFailedScaleset` would return the `ResourceNotFound` error. Then, `DeployRP` would log the `ResourceNotFound` error, and either continue (if there are still attempts left) or return the original deployment failure error.